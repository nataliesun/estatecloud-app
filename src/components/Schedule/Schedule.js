import React, { Component } from "react";
import DateApiService from '../../services/date-api-service';


import './Schedule.scss'
import SelectableCalendar from "../SelectableCalendar/SelectableCalendar";
import DateModal from "../DateModal/DateModal";
import DateContext from "../../contexts/DateContext";


class Schedule extends Component {
  static contextType = DateContext

  state = {
    modalClosed: true,
    recedeCalendar: false
  };

  handleBtnClick = () => {
    this.setState({
      modalClosed: !this.state.modalClosed,
      recedeCalendar: !this.state.recedeCalendar
    })
  }

  componentDidMount() {
    DateApiService.getDatesForProperty(this.props.match.params.property_id)
      .then(reservations => this.context.setDates(reservations))

  }

  render() {
    return (
      <div className="Schedule">
        <button id="Schedule__btn" onClick={this.handleBtnClick}>Add Dates</button>
        <DateModal hidden={this.state.modalClosed} handleClose={this.handleBtnClick} {...this.props} />
        <SelectableCalendar receded={this.state.recedeCalendar} />
      </div>
    );
  }
}

export default Schedule