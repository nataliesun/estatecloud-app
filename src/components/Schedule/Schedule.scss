@import '~react-big-calendar/lib/sass/styles';
@import "~react-big-calendar/lib/addons/dragAndDrop/styles.css";
@import "~react-big-calendar/lib/css/react-big-calendar.css";
@import '../variables.scss';

.Schedule {
  height: 60vh;

  #Schedule__btn {
    @include highlighted-button;
    margin-bottom: 20px;
    cursor: pointer;
  }

  @media screen and (max-width: $mobile-min-width) {
    display: flex;
    flex-direction: column;

    #Schedule__btn {
    @include highlighted-button;
    margin-bottom: 10px;
    cursor: pointer;
  }
  }
}