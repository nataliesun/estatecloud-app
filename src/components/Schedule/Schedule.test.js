import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Schedule from './Schedule';
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faDoorOpen, faPlusCircle, faPlus, faTimes
} from '@fortawesome/free-solid-svg-icons'

library.add(
  faDoorOpen, // logo
  faPlusCircle,
  faPlus,
  faTimes
)

describe(`Schedule Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Schedule />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Schedule />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})