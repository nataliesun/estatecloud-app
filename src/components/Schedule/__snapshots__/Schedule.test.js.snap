// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule Component renders the UI as expected 1`] = `
<div
  className="Schedule"
>
  <button
    id="Schedule__btn"
    onClick={[Function]}
  >
    Add Dates
  </button>
  <div
    className="closed DateModal"
  >
    <span
      className="close-modal"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <form
      onSubmit={[Function]}
    >
      <h3>
        New Reservation
      </h3>
      <input
        id="name"
        name="name"
        placeholder="Reservation title"
        required={true}
        type="text"
      />
      <div
        className="react-daterange-picker react-daterange-picker--closed react-daterange-picker--enabled"
        onFocus={[Function]}
      >
        <div
          className="react-daterange-picker__wrapper"
        >
          <div
            className="react-daterange-picker__inputGroup"
            onClick={[Function]}
            role="presentation"
          >
            <input
              max="275760-09-12"
              min="-271821-04-19"
              name="daterange_from"
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "left": "-9999px",
                  "position": "absolute",
                  "top": "-9999px",
                  "visibility": "hidden",
                }
              }
              type="date"
              value="2019-06-25"
            />
            
            <input
              autoComplete="off"
              className="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__month"
              max={12}
              min={1}
              name="month"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              placeholder="--"
              required={null}
              type="number"
              value={6}
            />
            <span
              className="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autoComplete="off"
              className="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__day"
              max={30}
              min={1}
              name="day"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              placeholder="--"
              required={null}
              type="number"
              value={25}
            />
            <span
              className="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autoComplete="off"
              className="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__year"
              max={275760}
              min={1000}
              name="year"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              placeholder="----"
              required={null}
              step={1}
              type="number"
              value={2019}
            />
            
          </div>
          â€“
          <div
            className="react-daterange-picker__inputGroup"
            onClick={[Function]}
            role="presentation"
          >
            <input
              max="275760-09-12"
              min="-271821-04-19"
              name="daterange_to"
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "left": "-9999px",
                  "position": "absolute",
                  "top": "-9999px",
                  "visibility": "hidden",
                }
              }
              type="date"
              value="2019-06-25"
            />
            
            <input
              autoComplete="off"
              className="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__month"
              max={12}
              min={1}
              name="month"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              placeholder="--"
              required={null}
              type="number"
              value={6}
            />
            <span
              className="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autoComplete="off"
              className="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__day"
              max={30}
              min={1}
              name="day"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              placeholder="--"
              required={null}
              type="number"
              value={25}
            />
            <span
              className="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autoComplete="off"
              className="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__year"
              max={275760}
              min={1000}
              name="year"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              placeholder="----"
              required={null}
              step={1}
              type="number"
              value={2019}
            />
            
          </div>
          <button
            className="react-daterange-picker__clear-button react-daterange-picker__button"
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                stroke="black"
                strokeWidth="2"
              >
                <line
                  x1="4"
                  x2="15"
                  y1="4"
                  y2="15"
                />
                <line
                  x1="15"
                  x2="4"
                  y1="4"
                  y2="15"
                />
              </g>
            </svg>
          </button>
          <button
            className="react-daterange-picker__calendar-button react-daterange-picker__button"
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                stroke="black"
                strokeWidth="2"
              >
                <rect
                  fill="none"
                  height="15"
                  width="15"
                  x="2"
                  y="2"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="0"
                  y2="4"
                />
                <line
                  x1="13"
                  x2="13"
                  y1="0"
                  y2="4"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <button
        id="submit-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <div
    className="SelectableCalendar"
    style={
      Object {
        "height": "70vh",
      }
    }
  >
    0
  </div>
  <div
    className="closed ReservationModal"
  >
    <span
      className="close-modal"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times fa-w-11 "
        data-icon="times"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 352 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <h3>
      
    </h3>
    <p>
      Info
    </p>
    <button
      id="submit-btn"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
